tpot/tpot.py

```diff
@@ -411,8 +403,8 @@ def _decision_tree(self, input_df, max_features, max_depth):
             max_features = 'auto'
         elif max_features == 1:
             max_features = None
-        elif max_features > len(input_df.columns) - 3:
-            max_features = len(input_df.columns) - 3
+        else:
+            max_features = min(max_features, len(input_df.columns) - 3)
```

Comment:
:+1: to changes like these where `if` branches are replaced with `min` and `max`. I would merge a PR with only those changes right now.